src/apps/answers/service.py

```diff
@@ -568,6 +581,48 @@ async def get_activity_answer(
         )
 
         return submission
+    
+    async def _validate_relation_between_subjects_in_applet(
+            self,
+            respondent_subject: SubjectSchema,
+            target_subject: SubjectSchema,
+            source_subject: SubjectSchema
+    ) -> None:
+        
+        """
+            Validate the relationship between subjects in an applet.
+            - respondent_subject: the subject inputting the answers.
+            - target_subject: the subject about whom the responses are.
+            - source_subject: the subject providing the responses.
+            
+            This method checks:
+            1. If there is a relationship between the logged-in user (respondent_subject) and target_subject.
+            2. If there is a relationship between the respondent_subject and the source_subject.
+            Raises:
+                MultiinformantAssessmentNoAccessApplet: If no valid relationship is found.
+        """
+
+        if respondent_subject.id == target_subject.id or respondent_subject.id == source_subject.id:
+            return None
+
+        relation_respondent_target_subjects_call = SubjectsCrud(self.session).get_relation(
+            respondent_subject.id, target_subject.id
+        )
+        relation_respondent_source_subjects_call = SubjectsCrud(self.session).get_relation(
+            respondent_subject.id, source_subject.id
+        )
+
+        relation_respondent_target_subjects, relation_respondent_source_subjects = await asyncio.gather(
+            relation_respondent_target_subjects_call,
+            relation_respondent_source_subjects_call
+        )
+        
+        if not relation_respondent_target_subjects and not relation_respondent_source_subjects:
```

Comment:
it still need to check if it is a temporary relation, get "meta" field and check for  "expires_at" date,  then validate this date. 

But this depends of other ticket that @sultanofcardio is working

Reply:
yes that is correct this is why we have the TODO comment there, the PR is just under review so I can understand what I am doing right and wrong in terms of code but It is still waiting for kenroy PR to be done to add the conditional logic here

Reply:
@sultanofcardio @rcmerlo is this check that we currently doing should be a "and" or an "or" ?

Reply:
It should be "or". It should fail if either relation is not present